web/packages/teleport/src/Users/UserList/UserList.tsx

```diff
@@ -16,52 +16,60 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import { Cell, LabelCell } from 'design/DataTable';
+import Table, { Cell, LabelCell } from 'design/DataTable';
 import { MenuButton, MenuItem } from 'shared/components/MenuAction';
+import { SearchPanel } from 'shared/components/Search';
 
-import { ClientSearcheableTableWithQueryParamSupport } from 'teleport/components/ClientSearcheableTableWithQueryParamSupport';
+import { SeversidePagination } from 'teleport/components/hooks/useServersidePagination';
 import { Access, User, UserOrigin } from 'teleport/services/user';
 
 export default function UserList({
-  users = [],
-  pageSize = 20,
   onEdit,
   onDelete,
   onReset,
+  onSearchChange,
+  search,
+  serversidePagination,
   usersAcl,
 }: Props) {
+  const canView = usersAcl.list && usersAcl.read;
+  const canEdit = usersAcl.edit;
+  const canDelete = usersAcl.remove;
+
   return (
-    <ClientSearcheableTableWithQueryParamSupport
-      data={users}
+    <Table
+      data={serversidePagination.fetchedData.agents}
+      fetching={{
+        fetchStatus: serversidePagination.fetchStatus,
+        onFetchNext: serversidePagination.fetchNext,
+        onFetchPrev: serversidePagination.fetchPrev,
+      }}
+      serversideProps={{
+        sort: undefined,
+        setSort: () => undefined,
```

Comment:
From what I see on the left, the previous view was sortable.

Reply:
It was, however this isn't feasible anymore with the paginated API since it doesn't support sorting, we ran into the same issue when converting the roles page to use the paginated API. We would need to implement a new sorted cache for users in the backend which is outside the scope of this PR.